### PR TITLE
[azure][feat] Add postgres service collection

### PR DESCRIPTION
**Pull request by 1101-1**
# Description
Add a postgresql service collection

# To-Dos

- [x] I have created tests for any new or updated functionality.
- [x] I ran `tox` successfully.

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://inventory.fix.security/code-of-conduct).

### CODE REVIEW

File: plugins/azure/fix_plugin_azure/resource/base.py [hunk 1]
```diff
@@ -473,6 +473,102 @@ class AzureUserAssignedIdentity:
...
+@define(eq=False, slots=False)
+class AzureServerBackup:
+    kind: ClassVar[str] = "azure_server_backup"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "backup_interval_hours": S("backupIntervalHours"),
+        "backup_retention_days": S("backupRetentionDays"),
+        "earliest_restore_date": S("earliestRestoreDate"),
+        "geo_redundant_backup": S("geoRedundantBackup"),
+    }
+    backup_interval_hours: Optional[int] = field(default=None, metadata={'description': 'Backup interval hours for the server.'})  # fmt: skip
```
Comment (aquamatthias) [inline plugins/azure/fix_plugin_azure/resource/base.py:510]: why is this in base?

File: plugins/azure/fix_plugin_azure/resource/base.py [hunk 2]
```diff
@@ -473,6 +473,102 @@ class AzureUserAssignedIdentity:
...
+
+@define(eq=False, slots=False)
+class AzureServerMaintenanceWindow:
+    kind: ClassVar[str] = "azure_server_maintenance_window"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "custom_window": S("customWindow"),
+        "day_of_week": S("dayOfWeek"),
+        "start_hour": S("startHour"),
+        "start_minute": S("startMinute"),
+    }
```
Comment (aquamatthias) [inline plugins/azure/fix_plugin_azure/resource/base.py:550]: base?

File: plugins/azure/fix_plugin_azure/resource/postgresql.py [hunk 3]
```diff
@@ -0,0 +1,641 @@
...
+                                    "memoryPerVCoreMb": sku.supported_memory_per_vcore_mb,
+                                },
+                                "storage": {
+                                    "iops": storage.supported_iops,
+                                    "storageSizeGb": (
+                                        storage.storage_size_mb // 1024 if storage.storage_size_mb else 0
+                                    ),
+                                },
+                                "location": self.location,
+                            }
```
Comment (aquamatthias) [inline plugins/azure/fix_plugin_azure/resource/postgresql.py:253]: We create here an arbitrary format which then is parsed in AzurePostgresqlServerType. 
Q1: what is the relationship between capability and server type? Can we get rid of one?
Q2: wouldn't it be simpler to create the server type based on the already parsed capability?

File: plugins/azure/fix_plugin_azure/resource/postgresql.py [hunk 4]
```diff
@@ -0,0 +1,641 @@
...
+
+            AzurePostgresqlServerType.collect(server_types, graph_builder)
+
+        if self.supported_psql_flexible_server_editions:
+            for edition in self.supported_psql_flexible_server_editions:
+                graph_builder.submit_work(service_name, collect_editions, edition)
+
+
+@define(eq=False, slots=False)
+class AzurePostgresqlServerConfiguration(MicrosoftResource, AzureProxyResource):
```
Comment (aquamatthias) [inline plugins/azure/fix_plugin_azure/resource/postgresql.py:264]: Same comment as on MySQL:
I would imagine something like this:

```python
class AzureDatabseConfiguration(MicrosoftResource):
   config: Optional[Json] = None
```

We would  read all ServerConfiguration Resources and turn them into a simple dict.

**Review by aquamatthias — CHANGES_REQUESTED** [review]
Thanks. Please find my comments.

File: plugins/azure/fix_plugin_azure/resource/base.py (hunk 2 above)
Comment (1101-1) [reply plugins/azure/fix_plugin_azure/resource/base.py:550]: We use `AzureServerBackup`, `AzureServerDataEncryption`, `AzureServerMaintenanceWindow`, `AzureServerHighAvailability` and `AzureServerNetwork` classes in `mysql` and `postgresql` resources.

File: plugins/azure/fix_plugin_azure/resource/postgresql.py (hunk 3 above)
Comment (1101-1) [reply plugins/azure/fix_plugin_azure/resource/postgresql.py:253]: We collect capabilities. Then from capabilities we create a `AzurePostgresqlServerType`/`AzureMysqlServerType`. After all works done, we remove capabilities from the graph.

Comment (aquamatthias) [reply plugins/azure/fix_plugin_azure/resource/postgresql.py:253]: Consider removing the capability class and read the related type directly.
This way we do not have an artificial type in our model, nor do we need to maintain 2 mappings.

File: plugins/azure/fix_plugin_azure/resource/base.py (hunk 2 above)
Comment (aquamatthias) [reply plugins/azure/fix_plugin_azure/resource/base.py:550]: I would try to keep the base as minimal as possible. Importing classes from MySQL is most probably fine in PostgreSQL (or vice versa) .

**Review by aquamatthias — APPROVED** [review]

**Pull request merged**